testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -1434,58 +1435,70 @@ def test_component_set_plural(self, attr, groupname):
-    
+
     @staticmethod
     @pytest.fixture()
     def universe():
         return make_Universe(extras=('masses', 'altLocs'))
 
+    @staticmethod
+    @pytest.fixture()
+    def atoms():
+        u = make_Universe(extras=("masses",), size=(3, 1, 1))
+        return u.atoms
+
     @pytest.mark.parametrize('attr', ['masses', 'altLocs'])
     def test_get_present_topattr_group(self, universe, attr):
         values = getattr(universe.atoms, attr)
         assert values is not None
-    
+
     @pytest.mark.parametrize('attr', ['mass', 'altLoc'])
     def test_get_present_topattr_component(self, universe, attr):
         value = getattr(universe.atoms[0], attr)
         assert value is not None
 
     @pytest.mark.parametrize('attr,singular', [
-        ('masses', 'mass'), 
+        ('masses', 'mass'),
         ('altLocs', 'altLoc')])
     def test_get_plural_topattr_from_component(self, universe, attr, singular):
         with pytest.raises(AttributeError) as exc:
             getattr(universe.atoms[0], attr)
         assert ('Do you mean ' + singular) in str(exc.value)
-    
+
     @pytest.mark.parametrize('attr,singular', [
-        ('masses', 'mass'), 
+        ('masses', 'mass'),
         ('altLocs', 'altLoc')])
     def test_get_sing_topattr_from_group(self, universe, attr, singular):
         with pytest.raises(AttributeError) as exc:
             getattr(universe.atoms, singular)
-        assert ('Do you mean '+attr) in str(exc.value)
-    
+        assert ('Do you mean ' + attr) in str(exc.value)
+
     @pytest.mark.parametrize('attr,singular', [
-        ('elements', 'element'), 
+        ('elements', 'element'),
         ('tempfactors', 'tempfactor'),
         ('bonds', 'bonds')])
     def test_get_absent_topattr_group(self, universe, attr, singular):
         with pytest.raises(NoDataError) as exc:
             getattr(universe.atoms, attr)
-        assert 'does not contain '+singular in str(exc.value)
+        assert 'does not contain ' + singular in str(exc.value)
 
     def test_get_non_topattr(self, universe):
         with pytest.raises(AttributeError) as exc:
             universe.atoms.jabberwocky
         assert 'has no attribute' in str(exc.value)
-
+        
     def test_unwrap_without_bonds(self, universe):
         with pytest.raises(NoDataError) as exc:
             universe.atoms.unwrap()
-        err = ('AtomGroup.unwrap() not available; '
-               'this requires Bonds')
-        assert str(exc.value) == err
+        
+        expected_message = (
+            "AtomGroup.unwrap() not available; this AtomGroup lacks defined bonds. "
+            "To resolve this, you can either:\n"
+            "1. Guess the bonds at universe creation using `guess_bonds = True`, or\n"
+            "2. Create a universe using a topology format where bonds are pre-defined."
+        )
+        expected_message_pattern = re.escape(expected_message)
+        assert re.fullmatch(expected_message_pattern, str(exc.value))
```

Comment:
Sorry, my previous comment wasn't clear enough. What I would like you to do is to

1. add to the `pytest.raises(NoDataError)` above a `match=expected_message_pattern`)
2. remove the `assert` that checks for the message.

In this way we have all checks inside the `pytest.raises`.